lib/evaluate/CMakeLists.txt

```diff
@@ -28,6 +28,7 @@ add_library(FortranEvaluate
   tools.cc
   type.cc
   variable.cc
+  rte.cc
```

Comment:
What does "rte" stand for?  Maybe "run-time environment"?  Perhaps "library" would be more clear.

Reply:
Yes it stands "run-time environment".  I agree "environment" is not appropriate and rte may be unclear. I find library also unclear on its own. I will go for "runtime-library". What about the following naming scheme:
`rte.cc -> runtime-library.cc`
`rte.h -> runtime-library-templates.h` (make it anoying to include because it should not be included if rte-interface.h is sufficient)
`rte-interface.h -> runtime-library.h`
For the namespace "rte", I don't know. Fortran::runtime is taken for the actual runtime implementation. Should I just remove the rte namespace for now?
Also, when they are many libraries described in runtime-library.cc, maybe they should go in different files of another folder, for now let's keep it simple.

Reply:
Regarding "make it annoying": In other projects, public/exported header files are placed in a directory (tree) and the `-I` option is used to select that directory.  This makes it easy to replace entire interfaces with no code changes. Meanwhile, header files that are privately shared within the implementation live in the same place as the source files.  Since module/library A only has access to its own directory and whatever lives at the `-I` directory (set), it can access module B's public interface but doesn't have backdoor access to module B's private parts.

Reply:
Why not, that would make things clear. I believe though that it is a repository wide decision and it should not be taken inside this PR. Let's discuss this with the others.

Reply:
Taking into consideration that everything that is done here is about intrinsic procedures, I have committed the following renaming in commit 7cfd33b:
 
rte-interface.h -> intrinsics-library.h
rte.h -> intrinsics-library-templates.h
rte.cc -> intrinsics-library.cc

`RteProcedureSymbol` becomes `IntrinsicProcedureRuntimeDescription`
`HostRteProcedureSymbol` becomes `HostRuntimeIntrinsicProcedure`
`TargetRteProcedureSymbol` becomes `TargetRuntimeIntrinsicProcedure`
`HostRte` becomes `HostIntrinsicProceduresLibrary`.
`TargetRteLibrary` becomes `TargetIntrinsicProceduresLibrary`.

the rte namespace is removed.